### PR TITLE
other/integeration_by_simpson_approx.py is added for approximate integeration

### PR DESCRIPTION
**Pull request by faizan2700**

### CODE REVIEW

**Comment by faizan2700** [comment]
Can anyone please tell me why my old pull request file changes (which were merged to master) are still being shown in files changed section the file I changed was iterating_through_submasks.py in older pull request. this pull request is for integeration_by_simpson_approx.py file and DIRECTORY.md changes accordingly. Thanks in advance

**Comment by cclauss** [comment]
CONTRIBUTING.md makes it clear that you should not modify DIRECTORY.md...  That file is autogenerated.

File: other/integeration_by_simpson_approx.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    >>> simpson_integration(lambda x : x*x,3.45,3.2,-1)
+    Traceback (most recent call last):
+        ...
+    AssertionError: precision should be positive integer your input : -1
+        
+    """
+    assert callable(
+        function
+    ), "the function(object) passed should be callable your input : {}".format(function)
```
Comment (cclauss) [inline other/integeration_by_simpson_approx.py]: Please use f"strings" as discussed in CONTRIBUTING.md

File: other/integeration_by_simpson_approx.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    h = (b - a) / N_STEPS
+    result = function(a) + function(b)
+
+    for i in range(1, N_STEPS):
+        a1 = a + h * i
+
+        if i % 2:
+            result += function(a1) * 4
+        else:
+            result += function(a1) * 2
```
Comment (cclauss) [inline other/integeration_by_simpson_approx.py]: result += function(a1) * (4 if i % 2 else 2)

**Pull request merged**